all tests are failing because of "Attach server listening on localhost:-1"

time="2017-04-13T21:06:26Z" level=info msg="Attach server listening on localhost:-1" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x999f4d]

goroutine 24 [running]:
github.com/vmware/vic/lib/portlayer/attach.TestAttachStartStop.func1()
	/drone/src/github.com/vmware/vic/lib/portlayer/attach/server_test.go:48 +0x16d
created by github.com/vmware/vic/lib/portlayer/attach.TestAttachStartStop
	/drone/src/github.com/vmware/vic/lib/portlayer/attach/server_test.go:68 +0x189
FAIL	github.com/vmware/vic/lib/portlayer/attach	0.012s
Makefile:261: recipe for target 'test-job-github.com/vmware/vic/lib' failed
make: *** [test-job-github.com/vmware/vic/lib] Error 1
[info] build failed (exit code 2)

--- a/lib/portlayer/attach/server_test.go
+++ b/lib/portlayer/attach/server_test.go
@@ -1,4 +1,4 @@
-// Copyright 2016 VMware, Inc. All Rights Reserved.
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -35,7 +35,7 @@ import (
 // Start the server, make 200 client connections, test they connect, then Stop.
 func TestAttachStartStop(t *testing.T) {
 	log.SetLevel(log.InfoLevel)
-	s := NewAttachServer("", -1)
+	s := NewAttachServer("", 0)
 
 	wg := &sync.WaitGroup{}
 
@@ -43,8 +43,9 @@ func TestAttachStartStop(t *testing.T) {
 		defer wg.Done()
 
 		c, err := net.Dial("tcp", s.l.Addr().String())
-		assert.NoError(t, err)
-		assert.NotNil(t, c)
+		if !assert.NoError(t, err) || !assert.NotNil(t, c) {
+			return
+		}
 		defer c.Close()
 
 		buf := make([]byte, 1)
@@ -88,7 +89,7 @@ func TestAttachStartStop(t *testing.T) {
 func TestAttachSshSession(t *testing.T) {
 	log.SetLevel(log.InfoLevel)
 
-	s := NewAttachServer("", -1)
+	s := NewAttachServer("", 0)
 	assert.NoError(t, s.Start(true))
 	defer s.Stop()
 

